cli/internal/cmd/apply.go

```diff
@@ -287,15 +284,15 @@ type applyCmd struct {
 
 	merger configMerger
 
-	applier applier
+	imageFetcher imageFetcher
+	applier      applier
 
 	newHelmClient   func(kubeConfigPath string, log debugLog) (helmApplier, error)
-	newDialer       func(validator atls.Validator) *dialer.Dialer
-	newKubeUpgrader func(out io.Writer, kubeConfigPath string, log debugLog) (kubernetesUpgrader, error)
 	newInfraApplier func(context.Context) (cloudApplier, func(), error)
 }
 
 type applier interface {
+	SetKubeConfig(kubeConfig []byte) error
```

Comment:
Not a fan of keeping separate clients. That is what we try to avoid with the Constellation library.
A "solution" could be passing the `kubeConfig` into every method which requires setting up a client, but this is also not necessarily better since it requires passing around the config everywhere.